tests/test_helpers/dataset_utils.py

```diff
@@ -0,0 +1,258 @@
+import re
+from typing import Any, Dict, Iterable, List, Optional
+
+import yaml
+
+from fidesops.graph.config import Collection, Field, FieldPath, ObjectField, ScalarField
+from fidesops.models.connectionconfig import ConnectionConfig
+from fidesops.models.datasetconfig import DatasetConfig, convert_dataset_to_graph
+from fidesops.schemas.dataset import FidesopsDataset
+from fidesops.util.collection_util import Row
+
+SAAS_DATASET_DIRECTORY = "data/saas/dataset/"
+
+
+def update_dataset(
+    connection_config: ConnectionConfig,
+    dataset_config: DatasetConfig,
+    api_data: Dict[str, List[Row]],
+    file_name: str,
+):
+    """
+    Helper function to update the dataset in the given dataset_config
+    with api_data and write the formatted result to the specified file.
+    """
+
+    generated_dataset = generate_dataset(
+        dataset_config.dataset,
+        api_data,
+        [endpoint["name"] for endpoint in connection_config.saas_config["endpoints"]],
+    )
+
+    # the yaml library doesn't allow us to just reformat
+    # the data_categories field so we fix it with a regex
+    #
+    # data_categories:
+    #   - system.operations
+    #
+    # data_categories: [system.operations]
+    #
+    with open(f"{SAAS_DATASET_DIRECTORY}{file_name}", "w") as dataset_file:
+        dataset_file.write(
+            re.sub(
+                r"(data_categories:)\n\s+- ([^\n]+)",
+                r"\1 [\2]",
+                yaml.dump(
+                    {"dataset": [generated_dataset]},
+                    default_flow_style=False,
+                    sort_keys=False,
+                    indent=2,
+                ),
+            )
+        )
+
+
+def generate_dataset(
+    existing_dataset: Dict[str, Any],
+    api_data: Dict[str, List[Row]],
+    collection_order: List[str] = None,
+):
+    """
+    Generates a dataset which is an aggregate of the existing dataset and
+    any new fields generated from the API data. Orders the collections
+    based on the order of collection_order.
+    """
+
+    # preserve the collection order in the dataset if a collection order is not provided
+    if not collection_order:
+        collection_order = [
+            collection["name"] for collection in existing_dataset["collections"]
+        ]
+
+    # remove the dataset name from the keys in the api_data map before passing
+    # into generate_collections
+    generated_collections = generate_collections(
+        {
+            collection_name.replace(f"{existing_dataset['fides_key']}:", ""): collection
```

Comment:
is this logic/processing covered in the unit tests? i know we're probably not going to be able to cover every piece of this, but it's some behavior that i think would be good to have covered both to ensure correct functionality and to clarify expected behavior

Reply:
also, i think my suggestion about an "integration" level test may help in providing coverage for this piece of things? that would address my concern, no need to make it specifically a unit test.

maybe just another reason to look into having that sort of "integration" test with more realistic data